jetty-core/jetty-client/src/main/java/org/eclipse/jetty/client/HttpReceiver.java

```diff
@@ -765,20 +763,43 @@ private void invokeDemandCallback(boolean invoke)
 
         @Override
         public void fail(Throwable failure)
+        {
+            boolean failed = error(failure);
+            if (failed)
+                HttpReceiver.this.failAndClose(failure);
+            invokeDemandCallback(true);
```

Comment:
Shouldn't `invokeDemandCallback(true)` only be called if `failed` is true?

Reply:
I would leave it in this way, as `invokeDemandCallback()` is "safe" in the sense that if there is no callback it's a no-op.